app/src/main/java/com/example/stretchy/features/createtraining/ui/CreateOrEditTrainingViewModel.kt

```diff
@@ -166,92 +158,106 @@ class CreateOrEditTrainingViewModel(
         _uiState.value = stateSuccess.copy(isAutomaticBreakButtonClicked = false)
     }
 
-    fun updateAutoBreakDuration(durationInSec: Int) = automaticBreakPreferences.updateAutoBreakDuration(durationInSec)
+    fun updateAutoBreakDuration(durationInSec: Int) =
+        automaticBreakPreferences.updateAutoBreakDuration(durationInSec)
 
-    private fun currentActivitySizeList(): Int =
-        (_uiState.value as? CreateTrainingUiState.Success)?.activities?.size ?: 0
+    fun getAutoBreakDuration(): Int = automaticBreakPreferences.getCurrentAutoBreakDuration()
 
-    private fun saveTraining() {
+    private fun saveTraining(activitiesList: List<Activity>) {
         viewModelScope.launch {
             val success = (_uiState.value as? CreateTrainingUiState.Success)
             if (success != null) {
                 repository.addTrainingWithActivities(
                     TrainingWithActivity(
                         success.currentName,
-                        TrainingType.STRETCH,
+                        trainingType,
                         true,
-                        success.activities
+                        activitiesList
                     )
                 )
                 _uiState.emit(CreateTrainingUiState.Done)
-            } else {
-                //todo toast
             }
         }
     }
 
-    private fun getCurrentActivities(state: CreateTrainingUiState.Success): MutableList<Activity> {
-        val currentList = mutableListOf<Activity>()
-        currentList.addAll(state.activities)
-        return currentList
-    }
-
     private fun isCreateTrainingButtonVisible(
```

Comment:
- [ ] check if this can be a view part
- [ ] check why or if the remember has to be a part of a certain composable or can be declared above it.